source/Ship.cpp

```diff
@@ -322,13 +322,19 @@ void Ship::Load(const DataNode &node)
 			if(child.HasChildren())
 				for(const DataNode &grand : child)
 				{
-					// Load in the effect(s) to be displayed when the ship launches.
+					// Load in the effect(s) to be displayed when the ship launches/boards.
 					if(grand.Token(0) == "launch effect" && grand.Size() >= 2)
 					{
 						int count = grand.Size() >= 3 ? static_cast<int>(grand.Value(2)) : 1;
 						const Effect *e = GameData::Effects().Get(grand.Token(1));
 						bay.launchEffects.insert(bay.launchEffects.end(), count, e);
 					}
+					else if(grand.Token(0) == "retrieve effect" && grand.Size() >= 2)
```

Comment:
This PR contains 2 changes; adding the `"retrieve effect"` and the functionality to let the carried ship dock on the bay location instead of the ships center.
I have the impression that the `"retrieve effect"` is a lot simpler to test and to review, so you might want to consider to put that in a separate PR (to try to get it merged ahead of the docking at the bay location).

Reply:
I can't. The retrieve effect is generated at the point where the carry docks at a specific bay, which I can't do without the rest of the PR. I could also remove it and then PR it later when this gets (hopefully) merged.

Reply:
Without this PR the game also knows at the moment of docking which bay is assigned to which carried ship (since the ship gets assigned to a bay at the moment of docking and drawn there (for external bays) from the moment of docking onward).

But I didn't look into the details and if this is difficult, then I'm also okay with just keeping it as part of this PR.

Reply:
Yes that is true. I'll clarify: I can make the effect happen on a specific bay, but then it looks off since carries still board the ship and not the dock.

Reply:
I agree, the effect works best if both PRs are merged.